src/renderer/html_handlebars/helpers/navigation.rs

```diff
@@ -5,165 +5,133 @@ use serde_json;
-                        let mut previous_chapter = BTreeMap::new();
-
-                        // Chapter title
-                        previous.get("name")
-                                .ok_or_else(|| {
-                                                RenderError::new("No title found for chapter in \
-                                                                  JSON data")
-                                            })
-                                .and_then(|n| {
-                                              previous_chapter.insert("title".to_owned(), json!(n));
-                                              Ok(())
-                                          })?;
-
-
-                        // Chapter link
-                        previous.get("path")
-                                .ok_or_else(|| {
-                                                RenderError::new("No path found for chapter in \
-                                                                  JSON data")
-                                            })
-                                .and_then(|p| {
-                            Path::new(p).with_extension("html")
-                                        .to_str()
-                                        .ok_or_else(|| {
-                                                        RenderError::new("Link could not be \
-                                                                          converted to str")
-                                                    })
-                                        .and_then(|p| {
-                                previous_chapter
-                                            .insert("link".to_owned(), json!(p.replace("\\", "/")));
-                                Ok(())
-                            })
-                        })?;
-
-
-                        debug!("[*]: Render template");
-                        // Render template
-                        _h.template()
-                          .ok_or_else(|| RenderError::new("Error with the handlebars template"))
-                          .and_then(|t| {
-                            let mut local_rc = rc.with_context(Context::wraps(&previous_chapter)?);
-                            t.render(r, &mut local_rc)
-                        })?;
+                if let Some(previous) = previous {
+                    if let Some(item) = target.find(&base_path, &path, &item, previous)? {
+                        return Ok(Some(item));
                     }
-                    break;
-                } else {
-                    previous = Some(item.clone());
                 }
+
+                previous = Some(item.clone());
             }
             _ => continue,
         }
     }
 
+   Ok(None)
+}
+
+fn render(
+    _h: &Helper,
+    r: &Handlebars,
+    rc: &mut RenderContext,
+    chapter: &StringMap,
+) -> Result<(), RenderError> {
```

Comment:
How hard would it be to write a test which checks this handlebars helper? Would it be possible to mock up a quick template then run the handlebars helper and assert that the rendered output looks like we expect it to?